rotkehlchen/chain/gnosis/transactions.py

```diff
@@ -11,13 +14,45 @@
 from .modules.xdai_bridge.constants import BLOCKREWARDS_ADDRESS
 
 if TYPE_CHECKING:
+    from rotkehlchen.chain.ethereum.types import LOG_ITERATION_CALLBACK_ARGUMENTS_TYPE
     from rotkehlchen.db.dbhandler import DBHandler
 
     from .node_inquirer import GnosisInquirer
 
 ADDED_RECEIVER_ABI: Final = [{'anonymous': False, 'inputs': [{'indexed': False, 'name': 'amount', 'type': 'uint256'}, {'indexed': True, 'name': 'receiver', 'type': 'address'}, {'indexed': True, 'name': 'bridge', 'type': 'address'}], 'name': 'AddedReceiver', 'type': 'event'}]  # noqa: E501
 DEPLOYED_BLOCK: Final = 9053325
 DEPLOYED_TS: Final = 1539027985
+NO_BLOCK_PROCESSED_VALUE = 0
+
+
+@dataclass
+class WithdrawalsQueryParameters:
+    last_block_processed: int
+    expected_topics: set[str]
+    gnosis_transactions: 'GnosisTransactions'
+
+
+def _process_withdrawals_events_cb(
+        last_block_queried: int,
+        filters: dict[str, Any],  # pylint: disable=unused-argument
+        new_events: list[dict[str, Any]],
+        cb_arguments: 'LOG_ITERATION_CALLBACK_ARGUMENTS_TYPE',
+) -> None:
+    """Callback that processes new events
+    This function also keeps track of the last block number processed using
+    the variable nonlocal variable last_block_processed. It is later used
+    in case of error to save the queried range.
+    """
+    assert cb_arguments is not None
+    cb_arguments.last_block_processed = last_block_queried
```

Comment:
Shouldn't this move at the end of the function? Since only then that's when the block is processed.